src/script/user/UserRepository.ts

```diff
@@ -536,18 +575,17 @@ export class UserRepository {
    * Detects if the user has a profile picture that uses the outdated picture API.
    * Will migrate the picture to the newer assets API if so.
    *
-   * @param {Object} userData - user data from the backend
-   * @returns {void}
+   * @param userData - user data from the backend
    */
-  _upgradePictureAsset(userData) {
+  _upgradePictureAsset<T extends UserUpdate>(userData: T): T {
     const hasPicture = userData.picture.length;
     const hasAsset = userData.assets.length;
 
     if (hasPicture) {
       if (!hasAsset) {
         // if there are no assets, just upload the old picture to the new api
         const {medium} = mapProfileAssetsV1(userData.id, userData.picture);
-        medium.load().then(imageBlob => this.change_picture(imageBlob));
+        medium.load().then(imageBlob => this.change_picture(imageBlob as Blob));
```

Comment:
Not sure if we should do a falsy check here since `load()` can return `void`.

Reply:
We probably handle the `undefined` case on the UI level... (just a guess).
I would not change anything as of now